cppcheck_junit.py

```diff
@@ -80,17 +86,23 @@ def parse_cppcheck(file_name: str) -> Dict[str, List[CppcheckError]]:
 
     errors = collections.defaultdict(list)
     for error_element in error_root:
-        location_element: ElementTree.Element = error_element.find("location")
-        if location_element is not None:
-            file = location_element.get("file")
-            line = int(location_element.get("line"))
-        else:
-            file = ""
-            line = 0
+        file = error_element.get("file0")
+        sources = []
+        for location in error_element.findall("location"):
+            if not file:
+                file = location.get("file")
+            sources.append(
+                (
+                    location.get("file"),
+                    int(location.get("line")),
+                    int(location.get("column")),
+                    location.get("info"),
+                )
+            )
```

Comment:
Previously, `file` attribute of a single `location` element was used. Now, the `file0` attribute of the `error` element is taken. I do not know of a case, but it is possible that multiple `location` entries can be from different files, and I think `file0` is the actual file being tested. If that is not available, then take the `file` attribute from a `location` element.